docs/source/configuration/slots.md

```diff
@@ -0,0 +1,379 @@
+
+# Slots
+
+Slots provide a way for Volto add-ons to insert their own components at predefined locations in the rendered page.
+
+```{note}
+This concept is inspired by the Plone Classic UI {doc}`plone:classic-ui/viewlets`.
+```
+
+
+## Anatomy
+
+Slots have a name, and they contain a list of slot components.
+
+Volto renders slots using the `SlotRenderer` component.
+You can add slot insertion points in your code, as shown in the following example.
+
+```ts
+<SlotRenderer name="toolbar" content={content} />
+```
+
+Slot components consist of a data structure with key/value pairs, where the keys are the parent's slot name, the slot component's name, the component to render in the slot, and optional predicates.
+They are registered in the {ref}`configuration registry using a specific API for slots <configuration-registry-for-slot-components>`.
+
+The renderer of a slot component is controlled by the presence or absence of a list of conditions called {term}`predicates`.
+
+You can register multiple slot components with the same name under the same slot, as long as they have different predicates or components.
+
+To illustrate how slots are structured and work, let's register a slot component, where the component is `@sneridagh what goes here?`, and the predicate matches a route that begins with `/de/about`.
+
+```ts
+config.registerSlotComponent({
+  slot: 'toolbar',
+  name: 'save',
+  component: '@sneridagh what goes here?',
+  predicates: [RouteCondition('/de/about')],
+});
+```
+
+The following tree structure diagram illustrates the resultant registration.
+
+```text
+Slot (`name`=`toolbar`)
+└── SlotComponent
+    ├── `slot`=`toolbar`
+    ├── `name`=`save`
+    ├── `component`=`@sneridagh what goes here?`
+    └── predicate of "only appear under `/de/about`"
+```
+
+Next, let's register another slot component in the same slot, with the same name and component, but with a different predicate where the content type matches either `Document` or `News Item`.
+
+```ts
+config.registerSlotComponent({
+  slot: 'toolbar',
+  name: 'save',
+  component: '@sneridagh what goes here?',
+  predicates: [ContentTypeCondition(['Document', 'News Item'])],
+});
+```
+
+The following tree structure diagram illustrates the result of the second registration.
+
+```text
+Slot (`name`=`toolbar`)
+├── SlotComponent
+│   ├── `slot`=`toolbar`
+│   ├── `name`=`save`
+│   ├── `component`=`@sneridagh what goes here?`
+│   └── predicate of "only appear under `/de/about`"
+└── SlotComponent
+    ├── `slot`=`toolbar`
+    ├── `name`=`save`
+    ├── `component`=`@sneridagh what goes here?`
+    └── predicate of "only appear when the content type is either a Document or News Item"
+```
+
+Finally, let's register another slot component in the same slot, with the same name, but with a different component and without a predicate.
+
+```ts
+config.registerSlotComponent({
+  slot: 'toolbar',
+  name: 'save',
+  component: '@sneridagh what goes here as a different component?',
```

Comment:
component: DefaultHeader

Reply:
Done in [40c1dfd](https://github.com/plone/volto/pull/5775/commits/40c1dfd3125bb2ee1594bffcef8560948bbc4130)

Reply:
Whoops, I mean [275b2c5](https://github.com/plone/volto/pull/5775/commits/275b2c53ea92a21037da01afbe35a4cd12e7a1c7)